From 14f6e03b0babf47a0ffadff91845a76237631ab6 Mon Sep 17 00:00:00 2001
From: Kreijstal <elektrischrainbow@gmail.com>
Date: Wed, 7 Oct 2026 14:56:43 +0200
Subject: [PATCH] cmake: define _USE_MATH_DEFINES for MinGW too

src/parse/utils/values.h uses M_PI. The MSVC branch already defines
_USE_MATH_DEFINES; mingw-w64's <math.h> has the same requirement under
-std=c++20 (strict ANSI), so the MinGW/GCC build failed with
"'M_PI' was not declared in this scope".
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba7c086..081d89d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -101,6 +101,9 @@ if (WIN32)
     else()
         # MinGW/GCC toolchain (Windows amd64)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-sign-compare -O3 ${ENV_ARCHFLAGS}")
+        # With -std=c++20 (strict ANSI) mingw-w64's <math.h> only defines M_PI
+        # when _USE_MATH_DEFINES is set, as with MSVC (src/parse/utils/values.h)
+        add_compile_definitions(_USE_MATH_DEFINES)
     endif()
 else()
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-sign-compare -O3 -DROOT_PATH='\"${TOPLEVEL_PREFIX_PATH}\"' ${ENV_ARCHFLAGS}")
-- 
2.55.0

